crates/agentgateway/src/llm/azureopenai.rs

```diff
@@ -0,0 +1,48 @@
+use agent_core::strng;
+use agent_core::strng::Strng;
+use bytes::Bytes;
+
+use super::universal;
+use crate::llm::AIError;
+use crate::*;
+
+#[apply(schema!)]
+pub struct Provider {
+	#[serde(default, skip_serializing_if = "Option::is_none")]
+	pub model: Option<Strng>, // this is the Azure OpenAI model deployment name
```

Comment:
So the model field is also required? (There's no default, right?)

Reply:
it's optional in the Provider because you can set it in the request.
If both the request and provider specify a model, the provider one takes precedence